src/MongoDB.Driver/ChangeStreamOptions.cs

```diff
@@ -26,11 +26,7 @@ public class ChangeStreamOptions
     {
         // private fields
         private int? _batchSize;
-        private Collation _collation;
-        private ChangeStreamFullDocumentOption _fullDocument = ChangeStreamFullDocumentOption.Default;
         private TimeSpan? _maxAwaitTime;
-        private BsonDocument _resumeAfter;
-        private BsonTimestamp _startAtOperationTime;
```

Comment:
Don't do this. It's fine to stick with explicit backing fields. There is no need to use automatic properties just because they exist.

Definitely let's never use mixed models, like this, where some of the state of the class is held in explicit backing fields and the rest of the state is held in automatic properties. That's just really confusing. 

After this change the risk is that one looks at this class and immediately assumes that the state of this class consists of only two fields: _batchSize and _maxAwaitTime. But that is wrong, the rest of the state now exists in automatic properties which can only be found by visually scanning for them.

Stick to explicit backing fields.

Reply:
To me having backing fields code at least for classes which provide only information about a business object (without business logic) makes the code unnecessarily bigger and difficult.

But Ok, I will change it.

Reply:
I'm sure reasonable people can have different opinions on when to use explicit backing fields or not.

But please, let's not MIX explicit backing fields and automatic properties in the same class. That's the worst possible combination.